tools/buildsys/src/deps.rs

```diff
@@ -0,0 +1,196 @@
+// TODO - fix all clippy issues
+#![allow(clippy::all)]
+
+use guppy::graph::{DependencyDirection, PackageGraph, PackageMetadata, PackageQuery};
+use guppy::{MetadataCommand, PackageId};
+use snafu::ResultExt;
+use std::error::Error;
+use std::fmt::{Debug, Display, Formatter};
+use std::fs;
+use std::path::{Path, PathBuf};
+use toml::Table;
+
+// TODO - move this to the Manifest struct
+/// List all packages that are package dependencies. That is, follow all dependencies in the cargo
+/// dependency graph that lead to more packages, and do not follow those that involve kits. This
+/// gives a list of all the packages that are required when we are build a package, or all of the
+/// packages that should be included when building a kit.
+pub fn package_dependencies(manifest_path: impl AsRef<Path>) -> Result<Vec<String>> {
+    let manifest_path = manifest_path.as_ref();
+    let (name, manifest_type) = get_name_and_type(&manifest_path).unwrap();
+    // TODO - deserialize instead of using a MetadataCommand.
+    let graph = MetadataCommand::new()
+        .manifest_path(&manifest_path)
+        .build_graph()
+        .unwrap();
```

Comment:
Seems like this might address the workspace / cargo.lock issue - if we run `cargo metadata` and point it at the `variants` workspace and just use that file here.